File: app/src/main/java/com/danielspeixoto/connect/view/activity/BaseActivity.java
package com.danielspeixoto.connect.view.activity;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import com.danielspeixoto.connect.R;
import com.danielspeixoto.connect.module.ActivityBase;
import butterknife.ButterKnife;

/**
 * Created by danielspeixoto on 13/11/16.
 */
public abstract class BaseActivity extends AppCompatActivity implements ActivityBase.View {

    protected static final String EMPTY_STRING = "";
    public final String TAG = getClass().getSimpleName();

    public void showMessage(String message) {
        Toast.makeText(getApplicationContext(), message, Toast.LENGTH_SHORT).show();
    }

    protected void onCreate(@Nullable Bundle savedInstanceState, int layout) {
        super.onCreate(savedInstanceState);
        setContentView(layout);
        ButterKnife.bind(this);
        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
    }

    @Override
    public BaseActivity getActivity() {
        return this;
    }

    public void goToActivity(Class clazz) {
        Intent intent = new Intent(this, clazz);
        startActivity(intent);
    }

    protected boolean checkTextEmpty(EditText editText) {
        return editText.getText().toString().equals(EMPTY_STRING);
    }

    protected String getText(TextView text) {
        return text.getText().toString().trim();
    }

    protected void clear(EditText editText) {
        editText.setText(EMPTY_STRING);
    }
}

File: app/src/main/java/com/danielspeixoto/connect/view/activity/BaseActivity.kt
package com.danielspeixoto.connect.view.activity

import android.app.ProgressDialog
import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.Gravity
import com.danielspeixoto.connect.R
import com.danielspeixoto.connect.contract.ActivityBase
import com.danielspeixoto.connect.util.PARAM_LAYOUT
import org.jetbrains.anko.*

/**
 * Created by danielspeixoto on 13/11/16.
 */
abstract class BaseActivity : AppCompatActivity(), ActivityBase.View {
    val TAG = javaClass.simpleName

    lateinit var loadingDialog : ProgressDialog

    override val activity: BaseActivity
        get() = this

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
    }

    override fun goToActivity(clazz: Class<*>) {
        val intent = Intent(this, clazz)
        startActivity(intent)
    }

    override fun showLoadingDialog() {
        loadingDialog = indeterminateProgressDialog(getString(R.string.loading))
        loadingDialog.setCancelable(false)
        loadingDialog.show()
    }

    override fun closeLoadingDialog() {
        if(loadingDialog != null) {
            loadingDialog.dismiss()
        }
    }

    override fun showSavedDialog(message : String) {
        showImageDialog(R.drawable.ic_assignment_ind, message)
    }

    override fun showErrorDialog() {
        showImageDialog(R.drawable.ic_report_problem, getString(R.string.error_occurred))
    }

    fun showImageDialog(drawable : Int, message: String) {
        alert {
            customView {
                verticalLayout {
                    lparams(width = wrapContent, height = wrapContent)
                    padding = PARAM_LAYOUT * 4
                    imageView {
                        imageResource = drawable

                    }
                    textView(message) {
                        textSize = (PARAM_LAYOUT * 3).toFloat()
                        gravity = Gravity.CENTER
                    }
                }
            }

        }.show()
    }
}
